libutil/dr_config.c

```diff
@@ -749,7 +769,7 @@ read_options(opt_info_t *opt_info, IF_REG_ELSE(ConfigGroup *proc_policy, FILE *f
      * approach would be to keep track of a string length and pass
      * that to get_next_token().
      */
-    len = MIN(DR_MAX_OPTIONS_LENGTH - 1, _tcslen(ptr));
+    len = strnlen(ptr, DR_MAX_OPTIONS_LENGTH);
```

Comment:
Wrong function for a wchar_t: Use _tcsnlen (and add it to our_tchar.h and possibly dr_frontend.h if desired)